pandas/_testing/asserters.py

```diff
@@ -322,7 +323,15 @@ def _check_types(left, right, obj: str = "Index") -> None:
     # skip exact index checking when `check_categorical` is False
     elif check_exact and check_categorical:
         if not left.equals(right):
-            mismatch = left._values != right._values
+            try:
+                mismatch = left._values != right._values
+            except TypeError:
```

Comment:
comment pointing back to the original issue, otherwise ill be confused as to why this can happen